Matplotlib does not build on OpenBSD using clang

### Bug report

**Bug summary**

On OpenBSD using the default compiler (clang), matplotlib currently does not build due to `cast from pointer to smaller type 'char'`.

**Code for reproduction**

```sh
export CC=clang CXX=clang++

python3 -m venv venv &&
. venv/bin/activate &&
pip install --upgrade pip &&
pip install --upgrade numpy cycler pyparsing python-dateutil pytz six kiwisolver &&
pip install --upgrade matplotlib >clang.log 2>&1
```

**Actual outcome**

```
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -fPIC -O2 -pipe -O2 -pipe -fPIC -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_ttconv_ARRAY_API -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -D__STDC_FORMAT_MACROS=1 -I/home/dahlbaek/test_matplotlib/venv/lib/python3.6/site-packages/numpy/core/include -Iextern -I/usr/local/include -I/usr/include -I. -I/home/dahlbaek/test_matplotlib/venv/include -I/usr/local/include/python3.6m -c extern/ttconv/pprdrv_tt.cpp -o build/temp.openbsd-6.3-amd64-3.6/extern/ttconv/pprdrv_tt.o
    extern/ttconv/pprdrv_tt.cpp:248:41: error: cast from pointer to smaller type 'char' loses information
                    font->Copyright[length]=(char)NULL;
                                            ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:264:42: error: cast from pointer to smaller type 'char' loses information
                    font->FamilyName[length]=(char)NULL;
                                             ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:280:37: error: cast from pointer to smaller type 'char' loses information
                    font->Style[length]=(char)NULL;
                                        ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:296:40: error: cast from pointer to smaller type 'char' loses information
                    font->FullName[length]=(char)NULL;
                                           ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:312:39: error: cast from pointer to smaller type 'char' loses information
                    font->Version[length]=(char)NULL;
                                          ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:328:40: error: cast from pointer to smaller type 'char' loses information
                    font->PostName[length]=(char)NULL;
                                           ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:343:42: error: cast from pointer to smaller type 'char' loses information
                    font->PostName[length/2]=(char)NULL;
                                             ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:358:41: error: cast from pointer to smaller type 'char' loses information
                    font->Trademark[length]=(char)NULL;
                                            ^~~~~~~~~~
    extern/ttconv/pprdrv_tt.cpp:1044:19: error: cast from pointer to smaller type 'char' loses information
            temp[len]=(char)NULL;   /* a buffer and make it ASCIIz. */
                      ^~~~~~~~~~
    9 errors generated.
    error: command 'clang' failed with exit status 1
```

For full output, see [clang.log](https://github.com/matplotlib/matplotlib/files/2128411/clang.log)


**Expected outcome**

Expected outcome is that matplotlib builds without complication.

**Matplotlib version**
<!--Please specify your platform and versions of the relevant libraries you are using:-->
  * Operating system: OpenBSD 6.3 amd64
  * Matplotlib version: 2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): NA
  * Python version: 3.6.5
  * Jupyter version (if applicable): NA
  * Other libraries:
```
$ clang -v
OpenBSD clang version 6.0.0 (tags/RELEASE_600/final) (based on LLVM 6.0.0)
Target: amd64-unknown-openbsd6.3
Thread model: posix
```

Python3 was installed using the OpenBSD package system,

    pkg_add python-3.6.5

<!--Please tell us how you installed matplotlib and python e.g., from source, pip, conda-->
<!--If you installed from conda, please specify which channel you used if not the default-->

--- a/extern/ttconv/pprdrv_tt.cpp
+++ b/extern/ttconv/pprdrv_tt.cpp
@@ -245,7 +245,7 @@ void Read_name(struct TTFONT *font)
             {
                 font->Copyright = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->Copyright,(const char*)strings+offset,length);
-                font->Copyright[length]=(char)NULL;
+                font->Copyright[length]='\0';
                 replace_newlines_with_spaces(font->Copyright);
 
 #ifdef DEBUG_TRUETYPE
@@ -261,7 +261,7 @@ void Read_name(struct TTFONT *font)
                 free(font->FamilyName);
                 font->FamilyName = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->FamilyName,(const char*)strings+offset,length);
-                font->FamilyName[length]=(char)NULL;
+                font->FamilyName[length]='\0';
                 replace_newlines_with_spaces(font->FamilyName);
 
 #ifdef DEBUG_TRUETYPE
@@ -277,7 +277,7 @@ void Read_name(struct TTFONT *font)
                 free(font->Style);
                 font->Style = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->Style,(const char*)strings+offset,length);
-                font->Style[length]=(char)NULL;
+                font->Style[length]='\0';
                 replace_newlines_with_spaces(font->Style);
 
 #ifdef DEBUG_TRUETYPE
@@ -293,7 +293,7 @@ void Read_name(struct TTFONT *font)
                 free(font->FullName);
                 font->FullName = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->FullName,(const char*)strings+offset,length);
-                font->FullName[length]=(char)NULL;
+                font->FullName[length]='\0';
                 replace_newlines_with_spaces(font->FullName);
 
 #ifdef DEBUG_TRUETYPE
@@ -309,7 +309,7 @@ void Read_name(struct TTFONT *font)
                 free(font->Version);
                 font->Version = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->Version,(const char*)strings+offset,length);
-                font->Version[length]=(char)NULL;
+                font->Version[length]='\0';
                 replace_newlines_with_spaces(font->Version);
 
 #ifdef DEBUG_TRUETYPE
@@ -325,7 +325,7 @@ void Read_name(struct TTFONT *font)
                 free(font->PostName);
                 font->PostName = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->PostName,(const char*)strings+offset,length);
-                font->PostName[length]=(char)NULL;
+                font->PostName[length]='\0';
                 replace_newlines_with_spaces(font->PostName);
 
 #ifdef DEBUG_TRUETYPE
@@ -340,7 +340,7 @@ void Read_name(struct TTFONT *font)
                 free(font->PostName);
                 font->PostName = (char*)calloc(sizeof(char),length+1);
                 utf16be_to_ascii(font->PostName, (char *)strings+offset, length);
-                font->PostName[length/2]=(char)NULL;
+                font->PostName[length/2]='\0';
                 replace_newlines_with_spaces(font->PostName);
 
 #ifdef DEBUG_TRUETYPE
@@ -355,7 +355,7 @@ void Read_name(struct TTFONT *font)
             {
                 font->Trademark = (char*)calloc(sizeof(char),length+1);
                 strncpy(font->Trademark,(const char*)strings+offset,length);
-                font->Trademark[length]=(char)NULL;
+                font->Trademark[length]='\0';
                 replace_newlines_with_spaces(font->Trademark);
 
 #ifdef DEBUG_TRUETYPE
@@ -1041,7 +1041,7 @@ const char *ttfont_CharStrings_getname(struct TTFONT *font, int charindex)
         }
 
         strncpy(temp,ptr,len);  /* Copy the pascal string into */
-        temp[len]=(char)NULL;   /* a buffer and make it ASCIIz. */
+        temp[len]='\0';   /* a buffer and make it ASCIIz. */
 
         return temp;
     }

